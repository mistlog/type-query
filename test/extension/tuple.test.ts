import { getResultCode } from "../common";

describe("tuple", () => {
    test("to object", () => {
        const name = "tuple-to-object";
        const code = getResultCode(name);
        expect(code).toMatchSnapshot();
    })
    
    test("first", () => {
        const name = "tuple-first";
        const code = getResultCode(name);
        expect(code).toMatchSnapshot();
    })

    test("length", () => {
        const name = "tuple-length";
        const code = getResultCode(name);
        expect(code).toMatchSnapshot();
    })

    test("push", () => {
        const name = "tuple-push";
        const code = getResultCode(name);
        expect(code).toMatchSnapshot();
    })

    test("includes", () => {
        const name = "tuple-includes";
        const code = getResultCode(name);
        expect(code).toMatchSnapshot();
    })

})