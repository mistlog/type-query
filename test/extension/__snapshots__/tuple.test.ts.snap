// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tuple first 1`] = `
"type arr1 = [\\"a\\", \\"b\\", \\"c\\"];
type head1 = \\"a\\";"
`;

exports[`tuple includes 1`] = `
"type input = [\\"a\\", \\"b\\", \\"c\\"];
type result = true;"
`;

exports[`tuple length 1`] = `
"type tesla = [\\"tesla\\", \\"model 3\\", \\"model X\\", \\"model Y\\"];
type teslaLength = 4;"
`;

exports[`tuple push 1`] = `
"type arr = [1, 2];
type result = [1, 2, \\"3\\"];"
`;

exports[`tuple to object 1`] = `
"type input = [\\"tesla\\", \\"model 3\\", \\"model X\\", \\"model Y\\"];
type result = {
  tesla: \\"tesla\\";
  \\"model 3\\": \\"model 3\\";
  \\"model X\\": \\"model X\\";
  \\"model Y\\": \\"model Y\\";
};"
`;
